rskj-core/src/main/java/org/ethereum/vm/program/Program.java

```diff
@@ -454,26 +470,46 @@ public void createContract(DataWord value, DataWord memStart, DataWord memSize)
         Repository track = getStorage().startTracking();
 
         //In case of hashing collisions, check for any balance before createAccount()
-        if (track.isExist(newAddress)) {
-            Coin oldBalance = track.getBalance(newAddress);
-            track.createAccount(newAddress);
-            track.addBalance(newAddress, oldBalance);
+        if (track.isExist(contractAddress)) {
```

Comment:
for CREATE2 collisions are possible see EIP-104
 * the code and nonce of the existing address must be checked and fail if non zero
 * the nonce of the `newAddress` must be incremented

Reply:
Ok, so the nonce increase of the `newAdress` would mean changing behavior of the `CREATE` opcode, because each new contract created would start with a non-zero nonce, we talked with @SergioDemianLerner and he decided it was better to keep compatibility with Ethereum so we are going to add the nonce increase, but only if the RSKIP is activated